androidApp/src/main/java/com/mbta/tid/mbta_app/android/nearbyTransit/NearbyTransitTabViewModel.kt

```diff
@@ -16,25 +17,29 @@ class NearbyTransitTabViewModel : ViewModel() {
 
     private val _currentNavEntry = MutableStateFlow<SheetRoutes?>(null)
     val currentNavEntry: StateFlow<SheetRoutes?> = _currentNavEntry
+    private val _previousNavEntry = MutableStateFlow<SheetRoutes?>(null)
+    val previousNavEntry: StateFlow<SheetRoutes?> = _previousNavEntry
 
     /**
-     * Record the current sheetRoute. Calling this function does *not* affect the navigation stack.
-     * It as a helper function to more conveniently access the SheetRoute directly rather than
-     * reading off `NavBackStackEntry`, where it may not always be known what SheetRoute type to
-     * resolve to.
+     * Record the current `navEntry` and archive the `previousNavEntry` Calling this function does
+     * *not* affect the navigation stack. It as a helper function to more conveniently access the
+     * SheetRoute directly rather than reading off `NavBackStackEntry`, where it may not always be
+     * known what SheetRoute type to resolve to.
      */
 
     // https://stackoverflow.com/a/78911122
     fun recordCurrentNavEntry(sheetRoute: SheetRoutes?) {
-        _currentNavEntry.value = sheetRoute
+
+        _currentNavEntry.update {
```

Comment:
TIL about [update](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/update.html) for atomically changing the value of state. This was more relevant for updating `stopData`, but also was convenient for updating the current nav entry & previous nav entry together.